### ISSUE TITLE
🐛 Bug Report: 

### ISSUE THREAD

**Issue by Sirfirelord69** [DiscordDungeons/Bugs#1104]
### 👟 Reproduction steps

check inventory
#+talk urist
get the dialogue
option 2 (upgrade weapon) 
(i chose iron sword, just to test)
number of upgrades
confirm dialogue blah
check inventory
check statsin case of auto equip

### 👍 Expected behavior

give me the upgraded weapon
(but also probably not equip it, check in case)

### 👎 Actual Behavior

doesnt give me the item

### 🎲 DRPG version

4.16.16-beta

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 📝 Additional context

#1067  bug still there
TL;DR buttons look nice 👍 
TTL;DR i bought TWO iron swords, so theres one missing in case yall wanna check from https://discord.com/channels/172382467385196544/891979582402605107/905436827987611678 and from https://discord.com/channels/172382467385196544/891979582402605107/905625592857243719 onwards @Sven65 @mou

**Comment by mount2010** [comment]
@Sven65 did you pull compile and update the ipc server

**Comment by mount2010** [comment]
Fixed
